criu/memfd.c

```diff
@@ -235,20 +227,58 @@ static struct memfd_inode *memfd_alloc_inode(int id)
 	return inode;
 }
 
-extern int restore_memfd_shm(int fd, u64 id, u64 size);
+static LIST_HEAD(inode_list);
```

Comment:
We already a list of inodes that we maintain during restore. Let's reuse it instead of creating a second one. Here's a commit that implement this idea: https://github.com/nviennot/criu/commit/c6ddea4eb119bde1e1e99eb81002ab2603eecfea

Note that the `struct memfd_inode` has a union for dump and restore parts. It doesn't really help us. The only common parts of the restore and dump inode are the list_head node, and the inode id. Might as well have two different structs `struct memfd_dump_inode` and `struct memfd_restore_inode`. Code would be clearer. What do you think?

Reply:
Thank you for the feedback. I addressed both your suggestions.